swift/ArrowFlight/Tests/ArrowFlightTests/FlightTest.swift

```diff
@@ -177,87 +187,104 @@ public class FlightClientTester {
         })
         
         XCTAssertEqual(actionTypes.count, 2)
-        XCTAssertEqual(actionTypes[0].type, "type1")
-        XCTAssertEqual(actionTypes[0].description, "desc1")
-        XCTAssertEqual(actionTypes[1].type, "type2")
-        XCTAssertEqual(actionTypes[1].description, "desc2")
+        
+        XCTAssertEqual(actionTypes[0].type, "clear")
+        XCTAssertEqual(actionTypes[0].description, "Clear the stored flights.")
+        XCTAssertEqual(actionTypes[1].type, "shutdown")
+        XCTAssertEqual(actionTypes[1].description, "Shut down this server.")
     }
     
     func listFlightsTest() async throws {
         let flightCriteria = FlightCriteria("flight criteria expression".data(using: .utf8)!)
         var num_calls = 0
         try await client?.listFlights(flightCriteria, closure: { data in
-            num_calls += 1
-            let schema = try streamToArrowSchema(data.schema)
-            XCTAssertEqual(schema.fields.count, 3)
+            if let schema = data.schema {
+                XCTAssertGreaterThanOrEqual(schema.fields.count, 0)
+                num_calls += 1
+            }
         })
         
-        XCTAssertEqual(num_calls, 1)
+        XCTAssertEqual(num_calls, 2)
     }
     
-    func doActionTest() async throws {
-        let action = FlightAction("test_action", body: "test_action body".data(using: .utf8)!)
+    func doActionTest(_ type: String, actionBody: Data) async throws {
+        let action = FlightAction(type, body: actionBody)
         var actionResults = [FlightResult]()
         try await client?.doAction(action, closure: { result in
             actionResults.append(result)
         })
         
-        XCTAssertEqual(actionResults.count, 1)
-        XCTAssertEqual(String(bytes:actionResults[0].body, encoding: .utf8), "test_action result")
+        XCTAssertEqual(actionResults.count, 0)
```

Comment:
It seems that this removes a test case for a `DoAction` that returns a result.
Is it intentional?

Reply:
This was intentional but I will put test case back.